OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -18,30 +20,35 @@ namespace OpenRA.Mods.Common.Activities
 	public class FlyAttack : Activity
 	{
 		readonly Target target;
+		readonly AttackPlane attackPlane;
+		readonly IEnumerable<AmmoPool> ammoPools;
 		Activity inner;
 		int ticksUntilTurn = 50;
 
-		public FlyAttack(Target target) { this.target = target; }
+		public FlyAttack(Actor self, Target target)
+		{
+			this.target = target;
+			attackPlane = self.TraitOrDefault<AttackPlane>();
+			ammoPools = self.TraitsImplementing<AmmoPool>();
+		}
 
 		public override Activity Tick(Actor self)
 		{
 			if (!target.IsValidFor(self))
 				return NextActivity;
 
-			var limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
-			if (limitedAmmo != null && !limitedAmmo.HasAmmo())
+			if (ammoPools != null && ammoPools.All(x => !x.Info.SelfReloads && !x.HasAmmo()))
```

Comment:
Maybe add a comment above this `if` for easier readability.
`// Move to the next activity only if all ammo pools are depleted and none reload automatically